index.js

```diff
@@ -180,7 +180,8 @@ class Peer extends stream.Duplex {
   }
 
   signal (data) {
-    if (this.destroyed) throw errCode(new Error('cannot signal after peer is destroyed'), 'ERR_SIGNALING')
+    if (this.destroying) return
+    if (this.destroyed) throw errCode(new Error('cannot signal after peer is destroyed'), 'ERR_DESTROYED')
```

Comment:
I don't like how there are now three behaviors for all public methods:
* Before `destroy` is called, calling a method has the desired effect
* Once `destroying` is set but before `destroyed` is set, calling a method does nothing
* Once `destroyed` is set it throws

Why not throw when either `destroying` or `destroyed` is set?

Reply:
On second thought, I can't think of any better options without breaking backwards compatibility.

At least the `error` event gets emitted, so the client should be cleaning everything up.